src/site-packages/openbmp/mrt2bmp/MrtProcessors.py

```diff
@@ -0,0 +1,753 @@
+
+        # Generate information TLVs about monitored router.
+        """
+         TLV Structure:
+
+          0                   1                   2                   3
+          0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         |          Information Type     |       Information Length      |
+         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         |                 Information (variable)                        |
+         ~                                                               ~
+         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+         Type = 0: String
+         Type = 1: sysDescr
+         Type = 2: sysName
+
+         sysDescr and sysName are must to sent.
+
+        """
+
+        # sysDescr tlv creation
+        f1 = '!H H 1s'
+        s1 = calcsize(f1)
+        sysDescr_data = pack(f1, 1, 1, ' ')
+
+        # sysName tlv creation
+        f2 = '!H H ' + str(len(self._router_name)) + 's'
+        s2 = calcsize(f2)
+        sysName_data = pack(f2, 2, len(self._router_name), self._router_name)
+
+        common_header = self.createBmpCommonHeader(3, s1 + s2 + 6, 4)
+
+        qm = common_header + sysDescr_data + sysName_data
+
+        return qm
+
+    def getTerminationMessage(self):
+
+        # Generate information TLVs about monitored router.
+        """
+         TLV Structure:
+
+          0                   1                   2                   3
+          0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         |          Information Type     |       Information Length      |
+         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+         |                 Information (variable)                        |
+         ~                                                               ~
+         +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+         Type = 0: String
+         Type = 1: reason
+
+        """
+
+        # Creation of Type 1 reason tlv
+        information = struct.pack("!H", 0)
+        reason_tlv = struct.pack("!H H", 1, len(information)) + information
+
+        common_header = self.createBmpCommonHeader(3, len(reason_tlv) + 6, 5)
+
+        qm = common_header + reason_tlv
+
+        return qm
+
+    def processRouteView(self):
+
+        if self._isToProcess:
+            for f in self._listOfRibAndUpdateFiles:
+
+                print "-- %s is started" % f[1]
+
+                if "rib" in f[1]:
+
+                    rp = RibProcessor(f[1], self._directory_path, self._router_name, self._collector_id, self._fwd_queue, self._log_queue)
+
+                    rp.processRibFile()
+
+                    moveFileToTempDirectory(os.path.join(self._directory_path, self._router_name, f[1]),
```

Comment:
Add try: except block here.  This is failing when directory doesn't exist.  Add logic in the config/startup to validate the paths correctly.